streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -538,13 +539,16 @@ public void flush() {
                 } catch (final RuntimeException exception) {
                     if (firstException == null) {
                         // do NOT wrap the error if it is actually caused by Streams itself
-                        if (exception instanceof StreamsException)
+                        // In case of FailedProcessingException Do not keep the failed processing exception in the stack trace
+                        if (exception instanceof FailedProcessingException)
+                            firstException = new StreamsException(exception.getCause());
```

Comment:
Before we introduced the `FailedProcessingException` we did not wrap the exception into a plain `StreamsException` but we wrapped it into a `ProcessorStateException` in the `else`-branch. We should keep that behavior.

Reply:
In case of FailedProcessingException, wrap it into a ProcessorStateException

```
 if (exception instanceof FailedProcessingException) {
    firstException = new ProcessorStateException(exception.getCause());
}
```